### ISSUE TITLE
Training data windowing?

### ISSUE THREAD

**Issue by JaeLee18** [amaiya/ktrain#392]
Hello,

With text_classifier using `bert`, I wonder is there any way to apply windowing training data, which were preprocessed by `texts_from_folder`.

Is this possible with `autofit` or `fit_onecycle`?

**Comment by amaiya** [comment]
Hi @JaeLee18 

If you're  referring to windowing as a [sampling technique ](http://www.cs.cmu.edu/afs/cs.cmu.edu/project/jair/pub/volume8/fuernkranz98a-html/node2.html), there is not built-in support for this, but you can easily implement this manually (using `texts_from_array` for example).

If you're referring to using a window to classify larger sequences with transformers models like BERT, there are a few ways to handle this.

1.   You can pre/post-process documents in the folder yourself to do what a sliding window does. Example: Split up documents in to subsequences with same label.  Classify document as mode of predictions on subsequences.
2.  Sample sentences from documents in one way or another  to make them smaller based on your specific applications.
3.  Try a transformer model like LongFormer that supports longer sequence sizes

#### LongFormer example
```python
# load text data
categories = ['alt.atheism', 'soc.religion.christian','comp.graphics', 'sci.med']
from sklearn.datasets import fetch_20newsgroups
train_b = fetch_20newsgroups(subset='train', categories=categories, shuffle=True)
test_b = fetch_20newsgroups(subset='test',categories=categories, shuffle=True)
(x_train, y_train) = (train_b.data, train_b.target)
(x_test, y_test) = (test_b.data, test_b.target)

# build and train Longformer model
import ktrain
from ktrain import text
MODEL_NAME = 'allenai/longformer-base-4096'
t = text.Transformer(MODEL_NAME, maxlen=1024, class_names=train_b.target_names)
trn = t.preprocess_train(x_train, y_train)
model = t.get_classifier()
learner = ktrain.get_learner(model, train_data=trn, batch_size=1)
learner.fit_onecycle(5e-5, 1)
```

Hope this helps.
